---
title: "MOMS-PI"
# author: "Author's Name"
date: "`r Sys.Date()`"
output:
  pdf_document:
    toc: no
  html_document:
    theme: cerulean
    toc: yes
---

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
library(dplyr)
options(stringsAsFactors = FALSE)
```

```{r}
library(readr)
library(biomformat)
library(ggplot2)
library(DataExplorer)
library(reshape2)
```

# Settings

```{r settings}
# Data folder
dataDir <- "/Users/mdozmorov/Documents/Data/GitHub/HMP2/iHMP2_Broad/"
```

# HMP2 Data

https://ibdmdb.org/tunnel/public/summary.html - 178 samples

## Metadata

```{r}

```
```{r}
# Input files
fileNameIn1 <- "hmp2_metadata.csv" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/Metadata/hmp2_metadata.csv", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_csv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_meta <- mtx %>% dplyr::select(-1) %>% colnames
samples_meta[1:20]
```

## 16S

### taxonomic_profiles.tsv.gz

```{r}
# Input files
fileNameIn1 <- "taxonomic_profiles.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/16S/1806/taxonomic_profiles.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_16S <- mtx %>% dplyr::select(-1) %>% colnames
```

## Serology

### hmp2_serology_Compiled_ELISA_Data.tsv

```{r}
# Input files
fileNameIn1 <- "hmp2_serology_Compiled_ELISA_Data.tsv"  
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/Serology/1740/hmp2_serology_Compiled_ELISA_Data.tsv", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_Serology <- mtx %>% dplyr::select(-1) %>% colnames
```

Overlap between samples

```{r}
dummy <- MDmisc::Venn2(samples_16S, samples_Serology, names = c("16S", "Serology"))
```


## Metagenomes

### taxonomic_profiles.tsv

```{r}
# Input files
fileNameIn1 <- "taxonomic_profiles_meta.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/WGS/1818/taxonomic_profiles.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_meta_taxonomic_profiles <- mtx %>% dplyr::select(-1) %>% colnames
```

### pathabundances.tsv

```{r}
# Input files
fileNameIn1 <- "pathabundances.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/WGS/1818/pathabundances.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_meta_pathabundances <- mtx %>% dplyr::select(-1) %>% colnames
```

### ecs.tsv

```{r}
# Input files
fileNameIn1 <- "ecs.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/WGS/1818/ecs.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_meta_ecs <- mtx %>% dplyr::select(-1) %>% colnames
```

Overlap between samples

```{r}
dummy <- MDmisc::Venn3(samples_meta_taxonomic_profiles, samples_meta_pathabundances, samples_meta_ecs, names = c("taxonomic_profiles", "pathabundances", "ecs"))
```

## Proteomics

### HMP2_proteomics_ecs.tsv

```{r}
# Input files
fileNameIn1 <- "HMP2_proteomics_ecs.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/Proteomics/1633/HMP2_proteomics_ecs.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_proteomics_ecs <- mtx %>% dplyr::select(-1) %>% colnames
```

### HMP2_proteomics_kos.tsv

```{r}
# Input files
fileNameIn1 <- "HMP2_proteomics_kos.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/Proteomics/1633/HMP2_proteomics_kos.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_proteomics_kos <- mtx %>% dplyr::select(-1) %>% colnames
```

Overlap between samples

```{r}
dummy <- MDmisc::Venn2(samples_proteomics_ecs, samples_proteomics_kos, names = c("ecs", "kos"))
dummy <- MDmisc::Venn2(samples_proteomics_ecs, samples_meta_taxonomic_profiles, names = c("ecs", "taxonomic_profiles"))
```

## Viromics

### taxonomic_profiles.tsv.gz

```{r}
# Input files
fileNameIn1 <- "taxonomic_profiles.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/Viromics/1732/taxonomic_profiles.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_viromics_taxonomic_profiles <- mtx %>% dplyr::select(-1) %>% colnames
```

### virome_virmap_analysis.tsv.gz

```{r}
# Input files
fileNameIn1 <- "virome_virmap_analysis.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/Viromics/1732/virome_virmap_analysis.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_viromics_virmap <- mtx %>% dplyr::select(-1) %>% colnames
```

Overlap between samples

```{r}
dummy <- MDmisc::Venn2(samples_viromics_taxonomic_profiles, samples_viromics_virmap, names = c("viromics_taxonomic_profiles", "virmap"))
dummy <- MDmisc::Venn2(samples_viromics_taxonomic_profiles, samples_16S, names = c("viromics_taxonomic_profiles", "16S"))
```


## Metabolites

### HMP2_metabolomics.csv.gz

```{r}
# Input files
fileNameIn1 <- "HMP2_metabolomics.csv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/Metabolites/1723/HMP2_metabolomics.csv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_csv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
# Sample names
samples_metabolites <- mtx %>% dplyr::select(-1) %>% colnames
```

## Metatranscriptomes

### genefamilies.tsv.gz

- Large file, 2164739 x 736
- Row names like "UniRef90_A0A009DJS4" "UniRef90_A0A009DJS4|unclassified" "UniRef90_A0A009DPX5"
- Column names like "CSM5FZ4M_Abundance-RPKs" "CSM5MCUO_Abundance-RPKs" "CSM5MCVN_Abundance-RPKs"

```{r eval=FALSE}
# Input files
fileNameIn1 <- "genefamilies.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/MTX/1750/genefamilies.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
```

### ecs.tsv.gz

```{r}
# Input files
fileNameIn1 <- "ecs_mtrans.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/MTX/1750/ecs.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
```

### pathabundances.tsv.gz

```{r}
# Input files
fileNameIn1 <- "pathabundances_mtrans.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/MTX/1750/pathabundances.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
```

## Host Transcriptomes

### host_tx_counts.tsv.gz

```{r}
# Input files
fileNameIn1 <- "host_tx_counts.tsv.gz" 
# Download the data if absent
if (!file.exists(paste0(dataDir, fileNameIn1))) {
  download.file(url = "https://ibdmdb.org/tunnel/products/HMP2/HTX/1730/host_tx_counts.tsv.gz", destfile = paste0(dataDir, fileNameIn1))
}
# Read in data
mtx <- read_tsv(paste0(dataDir, fileNameIn1))
print("Data dimensions")
dim(mtx)
# Peek into it
mtx[1:5, 1:5] %>% pander
```