---
title: "MOMS-PI data processing"
author: "Mikhail Dozmorov"
date: "`r Sys.Date()`"
output:
  pdf_document:
    toc: no
  html_document:
    theme: cerulean
    toc: yes
---

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
library(dplyr)
options(stringsAsFactors = FALSE)
```

```{r libraries}
library(phyloseq)
library(biomformat)
library(dplyr)
library(readr)
library(magrittr)
source("https://raw.githubusercontent.com/mdozmorov/MDmisc/master/R/venn.R")
load(file = "data/gg_13_5_taxonomy.rda") # mtx_tax_table for Greengenes, scripts/greengenes.R
```

```{r settings}
# Data folder
# dataDir <- "/Users/mdozmorov/Documents/Data/GitHub/HMP2/"
dataDir <- "D:/HMP2/"
# Use phyloseq or low-level reading of BIOM files
phyloseq_import_biom <- FALSE # If FALSE, `read_biom` will be used
# File names
fileNameIn1  <- "data.MOMS-PI/hmp_cart_16d395b515.tsv"          # Manifest
fileNameIn2  <- "data.MOMS-PI/hmp_cart_metadata_4568fb1008.tsv" # Metadata
fileNameOut1 <- "data.MOMS-PI/files_16S_unreadable.txt"         # Unreadable 16S files
# Files for matching filenames
fileNameOut2 <- "data.MOMS-PI/files_16S_actual_only.txt"        # Downloaded, but no annotations
fileNameOut3 <- "data.MOMS-PI/files_16S_listed_only.txt"        # Annotated, but not downloaded
fileNameOut4 <- "data.MOMS-PI/files_16S_actual_and_listed.txt"  # Downloaded AND Annotated
# Files for saving the results
fileNameOut5 <- "data.MOMS-PI/MOMSPI16S_mtx.rda"  # BIOM data
fileNameOut6 <- "data.MOMS-PI/MOMSPI16S_samp.rda" # Annotation data
fileNameOut7 <- "data.MOMS-PI/MOMSPI16S_tax.rda"  # Taxonomy data
```

```{r}
mtx1 <- read_tsv(fileNameIn1) # Manifest
mtx1 %<>% arrange(sample_id)  # Sort by sample_id
mtx2 <- read_tsv(fileNameIn2) # Metadata
mtx2 %<>% arrange(sample_id)  # Sort by sample_id
all.equal(mtx1$sample_id, mtx2$sample_id) # Should be TRUE

# Add file names
mtx1 %<>% mutate(file_name = sapply(mtx1$urls, function(x) strsplit(x, split = ",", fixed = TRUE)[[1]][1] %>% basename) %>% unname)
mtx1 <- left_join(mtx1, mtx2, by = "sample_id") # Join by sample_id
mtx1 <- mtx1[!duplicated(mtx1), ]                  # Remove duplicates

# Remove files that all have generic `otu_table.biom` name
# They are downloaded in the same folder and impossible to distinguish
indistinguishable_sample_id <- mtx1$sample_id[mtx1$file_name == "otu_table.biom"]
mtx1 <- mtx1[!(mtx1$file_name == "otu_table.biom"), ]
```

```{r}
# All downloaded files
files <- list.files(file.path(dataDir, "ptb"), pattern = "biom", full.names = TRUE)
# Set progress bar
pb <- txtProgressBar(min = 1, max = length(files), style = 3)

# Overlap between Downloaded (Actual) and Annotated (Listed) files
mtx_venn <- Venn2(files %>% basename, mtx1$file_name, c("Actual", "Listed"))
# Save the lists
writeLines(mtx_venn$id[mtx_venn$Actual == 1 & mtx_venn$Listed == 0 ], fileNameOut2) # Downloaded, but no annotations
writeLines(mtx_venn$id[mtx_venn$Actual == 0 & mtx_venn$Listed == 1 ], fileNameOut3) # Annotated, but not downloaded
writeLines(mtx_venn$id[mtx_venn$Actual == 1 & mtx_venn$Listed == 1 ], fileNameOut4) # Downloaded AND Annotated

# Match the order
mtx1 <- mtx1[match(basename(files), mtx1$file_name), ]
all.equal(basename(files), mtx1$file_name)
```

```{r eval = FALSE}
# Manually subset files and annotation for faster processing
files <- files[1:30]
mtx1  <- mtx1[1:30, ]
```

```{r}
my_import_biom <- function(file, phyloseq_import_biom = TRUE) {
  if (phyloseq_import_biom) { # phyloseq way
    mtx_biom <- import_biom(file) %>% otu_table() %>% as.data.frame() 
    mtx_biom <- data.frame(ID = rownames(mtx_biom), mtx_biom)
  } else { # bioformat way
    mtx_biom <- read_biom(file)
    mtx_biom_data <- biom_data(mtx_biom, parallel = TRUE) # Extract data one time 
    mtx_biom <- data.frame(ID = names(mtx_biom_data), mtx_biom_data)
  }
  return(mtx_biom)
}
```

```{r}
# Check if biom files in the specified folder can be read
unreadable_files <- list()
# Collect a list of unreadable files
for(i in 1:length(files)) {
  setTxtProgressBar(pb, i)
  res <- try(mtx_biom <- my_import_biom(files[i], phyloseq_import_biom), TRUE)
  if(isTRUE(class(res)=="try-error")) { 
    unreadable_files <- c(unreadable_files, basename(files[i])) 
  } else { next }
}
unreadable_files <- unlist(unreadable_files)

# Remove unreadable files from the lists
if(length(unreadable_files) > 0) {
  files <- files[ !(basename(files) %in% unreadable_files) ]
  mtx1  <- mtx1[  !(mtx1$file_name %in% unreadable_files), ]
  writeLines(unreadable_files, fileNameOut1) # Save the list of unreadable files
}
```

```{r}
# Read in all BIOM files
mtx_list <- list()
for(i in 1:length(files)) {
  setTxtProgressBar(pb, i)
  mtx_biom <- my_import_biom(files[[i]], phyloseq_import_biom)
  colnames(mtx_biom) <- c("ID", paste0('s', i))
  mtx_list <- c(mtx_list, list(mtx_biom))
}
# Combine all elements in a list into a data frame, merging by both IDs. Time consuming, over 5 hours
mtx_biom <- mtx_list %>% purrr::reduce(full_join, by = c("ID"))
mtx_biom[1:5, 1:5]

mtx_biom_rownames <- mtx_biom$ID  # Save ID column for rownames
mtx_biom <- mtx_biom[, -1]        # Remove ID column
mtx_biom[is.na(mtx_biom)] <- 0    # Replace NAs with zeros
mtx_biom <- as.matrix(mtx_biom)   # Convert to matrix
rownames(mtx_biom) <- mtx_biom_rownames  # Add rownames
colnames(mtx_biom) <- files %>% basename # Add colnames  
all.equal(colnames(mtx_biom), mtx1$file_name)
```


```{r}
# Check overlap between Greengenes IDs and BIOM IDs
# mtx_venn <- Venn2(rownames(mtx_tax_table), rownames(mtx_biom), c("Greengenes", "biom"))
setdiff(rownames(mtx_biom), rownames(mtx_tax_table)) # Should be 0
# Subset and match the taxonomy table to the BIOM table
mtx_tax_table <- mtx_tax_table[ rownames(mtx_tax_table) %in% rownames(mtx_biom), ]
mtx_tax_table <- mtx_tax_table[ match(rownames(mtx_biom), rownames(mtx_tax_table)), ]
all.equal(rownames(mtx_biom), rownames(mtx_tax_table))
```


```{r}
# Rename matrices for the package
momspi16S_mtx  <- mtx_biom
momspi16S_samp <- mtx1
momspi16S_tax  <- mtx_tax_table
# Make them compatible with the current objects in the R package
colnames(momspi16S_mtx) <- sub(pattern = ".otu_table.biom", replacement = "", x = colnames(momspi16S_mtx)) # Clean up sample names
momspi16S_samp <- as.data.frame(momspi16S_samp)
momspi16S_samp$file_name  <- sub(pattern = ".otu_table.biom", replacement = "", x = momspi16S_samp$file_name) # Clean up sample names
all.equal(colnames(momspi16S_mtx), momspi16S_samp$file_name)
rownames(momspi16S_samp) <- colnames(momspi16S_mtx)
# Save them
save(momspi16S_mtx,  file = fileNameOut5) # BIOM data
save(momspi16S_samp, file = fileNameOut6) # Annotation data
save(momspi16S_tax,  file = fileNameOut7) # Taxonomy data
```

Data dimensions, current

```{r echo=TRUE}
# Check dimensions
dim(momspi16S_mtx)
dim(momspi16S_samp)
dim(momspi16S_tax)
momspi16S_mtx[1:5, 1:5]
class(momspi16S_mtx)
momspi16S_samp[1:5, ]
class(momspi16S_samp)
momspi16S_tax[1:5, 1:5]
class(momspi16S_tax)
```

Data dimensions for the first version of the package

```
> # Check dimensions
> dim(momspi16S_mtx)
[1] 7665 9107
> dim(momspi16S_samp)
[1] 9107    9
> dim(momspi16S_tax)
[1] 7665    7
> momspi16S_mtx[1:5, 1:5]
        EP003595_K10_MV1D EP003595_K100_BRCD EP003595_K90_BCKD EP003595_K90_BRCD EP004835_K10_MCKD
100039                  0                  0                 0                 0                 0
1000547                 0                  0                 0                 0                 0
1005406                 0                  0                 0                 0                 0
1005533                 0                  0                 0                 0                 0
1007399                 0                  0                 0                 0                 0
> class(momspi16S_mtx)
[1] "matrix"
> momspi16S_samp[1:5, ]
                                          sample_id                       subject_id sample_body_site visit_number subject_gender
EP003595_K10_MV1D  8938ac880f194a32bc6b736c15714d95 858ed4564f11795ec13dda4c10a0d657           vagina            1         female
EP003595_K100_BRCD 8938ac880f194a32bc6b736c15717f36 858ed4564f11795ec13dda4c10a0d657           rectum           10         female
EP003595_K90_BCKD  8938ac880f194a32bc6b736c157154e4 858ed4564f11795ec13dda4c10a0d657    buccal mucosa            9         female
EP003595_K90_BRCD  8938ac880f194a32bc6b736c157164d6 858ed4564f11795ec13dda4c10a0d657           rectum            9         female
EP004835_K10_MCKD  e50d0c183689e4053ccb35f8b2de153a e50d0c183689e4053ccb35f8b29d98c8    buccal mucosa            1         female
                   subject_race study_full_name                         project_name               file
EP003595_K10_MV1D       unknown          momspi Integrative Human Microbiome Project  EP003595_K10_MV1D
EP003595_K100_BRCD      unknown          momspi Integrative Human Microbiome Project EP003595_K100_BRCD
EP003595_K90_BCKD       unknown          momspi Integrative Human Microbiome Project  EP003595_K90_BCKD
EP003595_K90_BRCD       unknown          momspi Integrative Human Microbiome Project  EP003595_K90_BRCD
EP004835_K10_MCKD       unknown          momspi Integrative Human Microbiome Project  EP004835_K10_MCKD
> class(momspi16S_samp)
[1] "data.frame"
> momspi16S_tax[1:5, 1:5]
        Kingdom    Phylum           Class                Order             Family             
100039  "Bacteria" "Firmicutes"     "Bacilli"            "Lactobacillales" "Enterococcaceae"  
1000547 "Bacteria" "Firmicutes"     "Bacilli"            "Lactobacillales" "Streptococcaceae" 
1005406 "Bacteria" "Firmicutes"     "Bacilli"            "Bacillales"      "Staphylococcaceae"
1005533 "Bacteria" "Firmicutes"     "Bacilli"            "Lactobacillales" "Carnobacteriaceae"
1007399 "Bacteria" "Proteobacteria" "Betaproteobacteria" "Neisseriales"    "Neisseriaceae"    
> class(momspi16S_tax)
[1] "matrix"
```

