---
title: "Exploratory 16S data analysis"
# author: "Author's Name"
date: "`r Sys.Date()`"
output:
  pdf_document:
    toc: no
  html_document:
    theme: cerulean
    toc: yes
---

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
library(dplyr)
options(stringsAsFactors = FALSE)
```

```{r libraries}
library(phyloseq)
library(ggplot2)
library(magrittr)
```


```{r settings}
# Big Data folder
dataDir      <- "~/Dropbox/MBWG/HMP2"

# Input files
fileNameIn  <- "hmp2_phyloseq_16S_momspi_KS.rda"     # All data from .biom files, merged by ID

```


# Load data

```{r}
load(file.path(dataDir, "data", fileNameIn))
mtx_phyloseq_16S
#replace na with a 0
otu_table(mtx_phyloseq_16S)[is.na(otu_table(mtx_phyloseq_16S))] <- 0
```

```{r, echo=TRUE}
names(sample_data(mtx_phyloseq_16S))

```

Merge taxa based on phylogenetic information

```{r, echo=TRUE}
#takes long time to run
#mtx_phyloseq_16S.genus = tax_glom(mtx_phyloseq_16S, "Genus")
#plot_tree(mtx_phyloseq_16S.genus, color="SampleType", shape="Class", size="abundance")
```


Select subsample of the data for 50 subjects

```{r, echo=TRUE}
ids <- sample(sample_data(mtx_phyloseq_16S)$subject_id, 50)

mtx_phyloseq_16S.sample <- subset_samples(mtx_phyloseq_16S, subject_id %in% ids)
mtx_phyloseq_16S.sample
```

Remove unobserved taxa

```{r, echo=TRUE}
mtx_phyloseq_16S.sample %<>%
    taxa_sums() %>%
    is_greater_than(0) %>%
    prune_taxa(mtx_phyloseq_16S.sample)
```


### Alpha Diversity Analysis

Alpha diversity measures the taxonomic variation within a sample and
`r BiocStyle::Biocpkg("phyloseq")` provides a method, `plot_richness`, to plot
various alpha diversity measures.

First a vector of richness (i.e. alpha diversity) measures is created to be
passed to the `plot_richness` method.

```{r, echo=TRUE}
richness_measures <-
    c("Observed", "Shannon", "Simpson")
```

```{r, echo=TRUE}
alpha_div <- estimate_richness(mtx_phyloseq_16S.sample, measures=richness_measures)
mtx_phyloseq_16S.sample %>%
    plot_richness(x = "sample_body_site", color = "sample_body_site", measures = richness_measures) +
    stat_boxplot(geom ="errorbar") +
    geom_boxplot() +
    theme_bw() +
    theme(axis.title.x = element_blank(), legend.position = "none")
```

### Principle Coordinates Analysis

```{r, echo=TRUE}
mtx_ordination <-
    ordinate(mtx_phyloseq_16S.sample, method = "PCoA", distance = "bray")

mtx_phyloseq_16S.sample %>%
    plot_ordination(mtx_ordination, color = "sample_body_site", shape = "sample_body_site") +
    theme_bw() +
    theme(legend.position = "bottom")
```


